44.md

```diff
@@ -0,0 +1,170 @@
+NIP-44
+======
+
+Encrypted Direct Message (Versioned)
+------------------------------------
+
+`optional` `author:paulmillr` `author:staab`
+
+The NIP introduces versioned encryption, allowing multiple algorithm choices to exist simultaneously.
+
+The algorithm described in NIP4 is potentially vulnerable to [padding oracle attacks](https://en.wikipedia.org/wiki/Padding_oracle_attack) and uses keys which are not indistinguishable from random.
+
+An encrypted payload MUST be encoded as a JSON object. Different versions may have different parameters. Every format has a `v` field specifying its version.
+
+Currently defined encryption algorithms:
+
+- `0x00` - Reserved
+- `0x01` - XChaCha with same key `sha256(ecdh)` per conversation
+
+# Version 0
+
+Version 0 is not defined, however implementations depending on this NIP MAY choose to support the payload described in NIP 04 in the same places a NIP 44 payload would otherwise be expected. This is intended to allow a smooth transition while clients and signing software adopt the new standard.
+
+# Version 1
+
+Params:
+
+1. `nonce`: base64-encoded xchacha nonce
+2. `ciphertext`: base64-encoded xchacha ciphertext, created from (key, nonce) against `plaintext`.
+
+Example:
+
+```
+{
+  "ciphertext": "FvQi1H4atMwU+FzUR/0CJ7kowjs+",
+  "nonce": "3dBKd83Pg2Q4Tu2A2e8N++c+ZW2IBc2f",
+  "v": 1
+}
+```
+
+**Note**: By default in the [libsecp256k1](https://github.com/bitcoin-core/secp256k1) ECDH implementation, the secret is the SHA256 hash of the shared point (both X and Y coordinates). We are using this exact implementation. In NIP4, unhashed shared point was used.
+
+## Code Samples
+
+### Javascript
+
+```javascript
+import {xchacha20} from "@noble/ciphers/chacha"
+import {secp256k1} from "@noble/curves/secp256k1"
+import {sha256} from "@noble/hashes/sha256"
+import {randomBytes} from "@noble/hashes/utils"
+import {base64} from "@scure/base"
+
+export const utf8Decoder = new TextDecoder()
+
+export const utf8Encoder = new TextEncoder()
+
+export const getSharedSecret = (privkey: string, pubkey: string): Uint8Array =>
+  sha256(secp256k1.getSharedSecret(privkey, "02" + pubkey).subarray(1, 33))
+
+export function encrypt(privkey: string, pubkey: string, text: string, v = 1) {
+  if (v !== 1) {
+    throw new Error("NIP44: unknown encryption version")
+  }
+
+  const key = getSharedSecret(privkey, pubkey)
+  const nonce = randomBytes(24)
+  const plaintext = utf8Encoder.encode(text)
+  const ciphertext = xchacha20(key, nonce, plaintext)
+
+  return JSON.stringify({
+    ciphertext: base64.encode(ciphertext),
+    nonce: base64.encode(nonce),
+    v,
+  })
```

Comment:
I'd prefer this so we can include it in at tag:

```suggestion
  return [base64.encode(ciphertext), base64.encode(nonce), v]
```

but if its too much bike shedding at this point than so be it.

Reply:
I'll resolve this conversation since it's a duplicate of https://github.com/nostr-protocol/nips/pull/716#discussion_r1291530129